### PR TITLE
Publish exsisting ports upon startup

### PR DESCRIPTION
**Pull request by Nino-K**
Signed-off-by: Nino Kodabande <nkodabande@suse.com>

### CODE REVIEW

File: pkg/docker/events.go [hunk 1]
```diff
@@ -54,6 +56,10 @@ func NewEventMonitor(portTracker *tracker.PortTracker) (*EventMonitor, error) {
 // MonitorPorts scans Docker's event stream API
 // for container start/stop events.
 func (e *EventMonitor) MonitorPorts(ctx context.Context) {
+	if err := e.initializeRunningContainers(ctx); err != nil {
+		log.Errorf("initializing exsisting container port mappings failed: %v", err)
```
Comment (mook-as) [inline pkg/docker/events.go]:
```suggestion
		log.Errorf("initializing existing container port mappings failed: %v", err)
```
Or a better rephrase:
```suggestion
		log.Errorf("failed to initialize existing container port mappings: %v", err)
```

File: pkg/docker/events.go [hunk 2]
```diff
@@ -103,3 +109,54 @@ func (e *EventMonitor) Info(ctx context.Context) error {
 
 	return err
 }
+
+func (e *EventMonitor) initializeRunningContainers(ctx context.Context) error {
+	containers, err := e.dockerClient.ContainerList(ctx, types.ContainerListOptions{
+		All:     true,
+		Filters: filters.NewArgs(filters.Arg("status", "running")),
```
Comment (mook-as) [inline pkg/docker/events.go]: I thought `All` was used to show stopped containers (and therefore setting it to false was the same as filtering by `status=running`)?

**Review by mook-as — CHANGES_REQUESTED** [review]

File: pkg/docker/events.go (hunk 2 above)
Comment (Nino-K) [reply pkg/docker/events.go]: You are right, it's not necessary.

**Review by mook-as — APPROVED** [review]

**Comment by Nino-K** [comment]
https://github.com/rancher-sandbox/rancher-desktop/issues/2677

**Pull request merged**